core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/version/VersionSearchRequest.java

```diff
@@ -112,7 +116,37 @@ protected Expression prepareQuery(RepositoryContext context) {
 
 		return query.build();
 	}
-
+	
+	protected final void addResourcesFilter(ExpressionBuilder queryBuilder, RepositoryContext context) {
+		User user = context.service(User.class);
+		
+		if (containsKey(OptionKey.RESOURCE)) {
+			Collection<String> resources = getCollection(OptionKey.RESOURCE, String.class);
+			Collection<String> resourceIds = resources.stream()
+					.map(resource -> resource.contains("/") ? new ResourceURI(resource).getResourceId() : resource)
+					.toList();
+			
+			if (user.isAdministrator()) {
```

Comment:
Not just the administrator user can access all resources, any user with the `browse:*` permission can do that, please fix and create test cases for the various scenarios. Thanks!